mava/wrappers/gigastep.py

```diff
@@ -0,0 +1,300 @@
+# Copyright 2022 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import TYPE_CHECKING, Dict, Tuple, Union
+
+import jax
+import jax.numpy as jnp
+from chex import Array, PRNGKey
+from gigastep.gigastep_env import GigastepEnv
+from jumanji import specs
+from jumanji.types import StepType, TimeStep, restart
+from jumanji.wrappers import Wrapper
+
+from mava.types import Observation, ObservationGlobalState, State
+
+if TYPE_CHECKING:  # https://github.com/python/mypy/issues/6239
+    from dataclasses import dataclass
+else:
+    from flax.struct import dataclass
+
+
+@dataclass
+class GigastepState:
+    """Wrapper around a Gigastep state to provide necessary attributes for Jumanji environments."""
+
+    state: State
+    key: PRNGKey
+    step: int
+    adversary_action: Array
+
+
+class GigastepWrapper(Wrapper):
```

Comment:
I know I told you to move this, but can we add just the top line back here please :smile: 

```suggestion
class GigastepWrapper(Wrapper):
        """Wraps a Gigastep environment so that its API is compatible with Jumanji environments."""
```